scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -560,7 +565,7 @@ class ScioContext private[scio] (val options: PipelineOptions, private var artif
   // =======================================================================
 
   /**  Whether this is a test context. */
-  def isTest: Boolean = testId.isDefined
+  lazy val isTest: Boolean = testId.isDefined
```

Comment:
@ClaireMcGinty I think it's ok to keep it as `def` there's no point in having it as `lazy val`